packages/edit-site/src/components/post-fields/index.js

```diff
@@ -284,12 +284,16 @@ function usePostFields( viewType ) {
 				label: __( 'Status' ),
 				id: 'status',
 				type: 'text',
-				elements: STATUSES,
+				elements: STATUSES.filter( ( { value } ) => value !== 'trash' ),
```

Comment:
For me it feels weird to kind of duplicate the "elements" property. Also "trash" is a valid value of "status" so I don't think we should be removing it from the "elements" of the field.

Are there any alternatives we can think about than having to duplicate the elements?

Reply:
There are the requisites, how filtering/editing differ:

- description is a valid option value. In some filters we may actually want to use it if present. In this particular case, we don't and that's different from the edit experience. We cannot just ignore description in the filter.
- the set of valid statuses for filtering and editing is different: in editing we don't want trash.

In most cases, I presume we'd only need `field.elements`, so I personally like this approach — it provides some flexibility and it's simple.

Thought a bit about "configuring" elements for the different contexts but I didn't come up with anything I liked more than this.

Reply:
What you're suggesting is that:

 - field.elements is for editing (DataForm) and the default for filters
 - field.filterBy.elements is for overriding for filters
 - There's nothing to override for "data form".

So the assumption is that "editing" is the main thing for elements. It bothers me a bit that we're treating DataForms and DataViews differently. Also for me, when we define a field, we should ideally detach ourself a bit from how it's used. So I don't really see why "trash" shouldn't be present in `field.elements` it's a valid status for that particular field.

One option I suggested before for the "editing" context, is to mark some elements as `readonly`. 

For the "description" I don't have a good solution either, but I wonder why we want to show the descriptions just for editing and not for filtering. If these are "unclear", they should be clarified in all parts no?

Reply:
>Thought a bit about "configuring" elements for the different contexts but I didn't come up with anything I liked more than this.

What did you try here and you didn't like? For `elements` to have an optional `context`(`edit|view`) prop?